File: com.teaminabox.eclipse.sortit/src/com/teaminabox/eclipse/sortit/LengthSortAction.scala
package com.teaminabox.eclipse.sortit

import org.eclipse.jface.viewers.ISelection
import org.eclipse.jface.action.IAction
import org.eclipse.ui.IWorkbenchWindow

class LengthSortAction extends SortItAction {
  def getComparator() = (a: String, b: String) => if (a.length == b.length) a < b else a.length < b.length
}

File: com.teaminabox.eclipse.sortit/src/com/teaminabox/eclipse/sortit/LengthSortAction.java
package com.teaminabox.eclipse.sortit;

import java.util.Comparator;

public class LengthSortAction extends SortItAction {
	public Comparator<String> getComparator() {
		return new Comparator<String>() {
			@Override
			public int compare(String a, String b) {
				if (a.length() == b.length()) {
					return a.compareTo(b);
				}
				return a.length() - b.length();
			}
		};
	}
}